source/vir/src/sst_elaborate.rs

```diff
@@ -214,6 +247,65 @@ impl<'a, 'b, D: Diagnostics> Visitor<Rewrite, VirErr, NoScoper> for ElaborateVis
     }
 }
 
+struct ElaborateVisitorBv<'a> {
+    ctx: &'a Ctx,
+    fun_ssts: SstMap,
+}
+
+impl<'a> ElaborateVisitorBv<'a> {
+    fn expand(&self, exps: Vec<Exp>) -> Result<Vec<Exp>, VirErr> {
+        vec_map_result(&exps, |e| {
+            crate::interpreter::eval_expr(
+                &self.ctx.global,
+                e,
+                None::<&air::messages::Reporter>,
+                self.fun_ssts.clone(),
+                self.ctx.global.rlimit,
+                self.ctx.global.arch,
+                crate::ast::ComputeMode::Z3,
+                &mut self.ctx.global.interpreter_log.lock().unwrap(),
+            )
+        })
+    }
+}
+
+impl<'a> Visitor<Rewrite, VirErr, NoScoper> for ElaborateVisitorBv<'a> {
+    // This is the same as visit_func_check in sst_visitor.rs, except for the call to self.expand(..)
```

Comment:
Do you need to override visit_func_check and visit_postcondition?  Would overriding just visit_exp be enough?

Reply:
Good question.  Based on one of my earlier attempts, I think overriding just `visit_exp` might do the wrong thing.  In particular, earlier I also overrode `FuncDeclSst`, so that I could expand function calls in its pre-/post-conditions.  However, when I did that, we lost information in the emitted assumption for the postcondition.  For example, instead of learning:
```
forall|a: u32| #[trigger] (a & 0) == 0
```
we woud learn:
```
forall|a: u32| #[trigger] 0 == 0
```
because the interpreter simplified the inner expression.

Given that my override of `FuncDeclSst` did get called and had this effect, I'm guessing that if we universally override `visit_exp`, then I think we'd see the same effect.  Does that seem accurate?